Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -1016,13 +1017,220 @@ def set_user_command(args):
+        'Description': response.get('description', ''),
+        'Addresses': response.get('addresses', []),
+        'IpCategories': response.get('ipCategories', []),
+        'Countries': response.get('countries', []),
+        'IsNonEditable': response.get('isNonEditable', False)
+    }
+    entry = {
+        'Type': entryTypes['note'],
+        'Contents': content,
+        'ContentsFormat': formats['json'],
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("IP Destination group created",
+                                         content),
+        'EntryContext': {'Zscaler.IPDestinationGroup': content},
+    }
+    return entry
+
+
+def list_ip_destination_groups(args: dict):
+    ip_group_id = str(args.get('ip_group_id', '')).strip()
+    exclude_type = str(args.get('exclude_type', '')).strip()
+    include_ipv6 = argToBoolean(args.get('include_ipv6', False))
+    if ip_group_id == '':
+        if include_ipv6:
+            try:
+                ipv4_cmd_url = "/ipDestinationGroups"
+                ipv6_cmd_url = "/ipDestinationGroups/ipv6DestinationGroups"
+                if exclude_type != '':
+                    ipv4_cmd_url += f"?excludeType={exclude_type}"
+                    ipv6_cmd_url += f"?excludeType={exclude_type}"
+                ipv4_responses = http_request('GET', ipv4_cmd_url).json()
+                ipv6_responses = http_request('GET', ipv6_cmd_url).json()
+            except json.decoder.JSONDecodeError as exp:
+                return_error(f'Failed to execute zscaler-list-ip-destination-group command. Error: {str(exp)}')
+            ipv4_contents = list()
+            ipv6_contents = list()
+            for response in ipv4_responses:
+                content = {
+                    'ID': int(response.get('id', '')),
+                    'Name': response.get('name', ''),
+                    'Type': response.get('type', ''),
+                    'Description': response.get('description', ''),
+                    'Addresses': response.get('addresses', []),
+                    'IpCategories': response.get('ipCategories', []),
+                    'Countries': response.get('countries', [])
+                }
+                ipv4_contents.append(content)
+            for response in ipv6_responses:
+                content = {
+                    'ID': int(response.get('id', '')),
+                    'Name': response.get('name', ''),
+                    'Type': response.get('type', ''),
+                    'Description': response.get('description', ''),
+                    'Addresses': response.get('addresses', []),
+                    'IpCategories': response.get('ipCategories', []),
+                    'Countries': response.get('countries', [])
+                }
+                ipv6_contents.append(content)
+            s_content = {
+                "IPv4": ipv4_contents,
+                "IPv6": ipv6_contents
+            }
```

Comment:
You can use a function to reduce the repetition of code. Here's an example of how you can refactor the code block:
```py
def get_contents(responses, type):
    contents = list()
    for response in responses:
        content = {
            'ID': int(response.get('id', '')),
            'Name': response.get('name', ''),
            'Type': response.get('type', ''),
            'Description': response.get('description', ''),
            'Addresses': response.get('addresses', []),
            'IpCategories': response.get('ipCategories', []),
            'Countries': response.get('countries', [])
        }
        contents.append(content)
    return {type: contents}
ipv4_contents = get_contents(ipv4_responses, "IPv4")
ipv6_contents = get_contents(ipv6_responses, "IPv6")
s_content = {
    **ipv4_contents,
    **ipv6_contents
}
```

Reply:
Good suggestion. I totally missed this idea. :) done